core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -933,17 +933,26 @@ object ZSTM {
    * Atomically performs a batch of operations in a single transaction.
    */
   def atomically[R, E, A](stm: ZSTM[R, E, A]): ZIO[R, E, A] =
-    ZIO.accessM[R] { r =>
-      ZIO.effectSuspendTotalWith { (platform, fiberId) =>
-        tryCommit(platform, fiberId, stm, r) match {
-          case TryCommit.Done(io) => io // TODO: Interruptible in Suspend
-          case TryCommit.Suspend(journal) =>
-            val txnId     = makeTxnId()
-            val done      = new AtomicBoolean(false)
-            val interrupt = UIO(Sync(done)(done.set(true)))
-            val async     = ZIO.effectAsync(tryCommitAsync(journal, platform, fiberId, stm, txnId, done, r))
-
-            async ensuring interrupt
+    ZIO.uninterruptibleMask { restore =>
+      ZIO.accessM[R] { r =>
+        ZIO.effectSuspendTotalWith { (platform, fiberId) =>
+          val state = new AtomicReference[State[E, A]](State.Running)
```

Comment:
The state is always created even in the "happy path", which degrades performance of all code.

Perhaps `TryCommit.Suspend` could return the atomic reference, i.e. push its allocation into the non-happy path (async STM).

Reply:
I created a `tryCommitSync` variant that doesn't require the state so that we could push its creation into the asynchronous path.